sources/Renderer/OpenGL/Platform/Linux/LinuxGLSwapChainContext.cpp

```diff
@@ -22,61 +23,151 @@ namespace LLGL
+    }
+    else
+    {
+        return MakeUnique<LinuxX11GLSwapChainContext>(static_cast<LinuxGLContextX11&>(context), surface);
+    }
+#else
+    return MakeUnique<LinuxX11GLSwapChainContext>(static_cast<LinuxGLContextX11&>(context), surface);
+#endif
 }
 
 bool GLSwapChainContext::MakeCurrentUnchecked(GLSwapChainContext* context)
 {
-    return LinuxGLSwapChainContext::MakeCurrentGLXContext(static_cast<LinuxGLSwapChainContext*>(context));
+#if LLGL_LINUX_ENABLE_WAYLAND
+    LinuxGLContext& linuxContext = LLGL_CAST(LinuxGLContext&, context->GetGLContext());
+
+    if (linuxContext.IsWayland())
+        return LinuxWaylandGLSwapChainContext::MakeCurrentEGLContext(static_cast<LinuxWaylandGLSwapChainContext*>(context));
+    else
+        return LinuxX11GLSwapChainContext::MakeCurrentGLXContext(static_cast<LinuxX11GLSwapChainContext*>(context));
+#else
+    return LinuxX11GLSwapChainContext::MakeCurrentGLXContext(static_cast<LinuxX11GLSwapChainContext*>(context));
+#endif
 }
 
 
 /*
- * LinuxGLSwapChainContext class
+ * LinuxX11GLSwapChainContext class
  */
 
-LinuxGLSwapChainContext::LinuxGLSwapChainContext(LinuxGLContext& context, Surface& surface) :
+LinuxX11GLSwapChainContext::LinuxX11GLSwapChainContext(LinuxGLContextX11& context, Surface& surface) :
     GLSwapChainContext { context                 },
-    glc_               { context.GetGLXContext() }
+    glc_               { static_cast<GLXContext>(context.GetGLXContext()) }
 {
     /* Get native window handle */
     NativeHandle nativeHandle = {};
     if (surface.GetNativeHandle(&nativeHandle, sizeof(nativeHandle)))
     {
-        dpy_ = nativeHandle.display;
-        wnd_ = nativeHandle.window;
+        dpy_ = nativeHandle.x11.display;
+        wnd_ = nativeHandle.x11.window;
     }
     else
         LLGL_TRAP("failed to get X11 Display and Window from swap-chain surface");
 }
 
-bool LinuxGLSwapChainContext::HasDrawable() const
+bool LinuxX11GLSwapChainContext::HasDrawable() const
 {
     return (wnd_ != 0);
 }
 
-bool LinuxGLSwapChainContext::SwapBuffers()
+bool LinuxX11GLSwapChainContext::SwapBuffers()
 {
     glXSwapBuffers(dpy_, wnd_);
     return true;
 }
 
-void LinuxGLSwapChainContext::Resize(const Extent2D& resolution)
+void LinuxX11GLSwapChainContext::Resize(const Extent2D& resolution)
 {
     // dummy
 }
 
-bool LinuxGLSwapChainContext::MakeCurrentGLXContext(LinuxGLSwapChainContext* context)
+bool LinuxX11GLSwapChainContext::MakeCurrentGLXContext(LinuxX11GLSwapChainContext *context)
 {
     if (context)
         return glXMakeCurrent(context->dpy_, context->wnd_, context->glc_);
     else
         return glXMakeCurrent(nullptr, 0, 0);
 }
 
+} // /namespace LLGL
+
+/*
+ * LinuxWaylandGLSwapChainContext class
+ */
+
+#if LLGL_LINUX_ENABLE_WAYLAND
+
+#include "LinuxGLContextWayland.h"
+
+#include <wayland-egl-core.h>
+
+namespace LLGL {
+
+LinuxWaylandGLSwapChainContext::LinuxWaylandGLSwapChainContext(LinuxGLContextWayland& context, Surface& surface) :
```

Comment:
I'm glad to see you already split up `LinuxGLSwapChainContext` into two separate classes but can we also split the file the same way you did with _LinuxGLContext.cpp_? And can we use the same name pattern you used for `LinuxGLContext`, i.e. use `LinuxGLSwapChainContextWayland`/`X11`?